### PR TITLE
Limit group validation to the used providers

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/core/externalvolume/VolumesModule.scala [hunk 1]
```diff
@@ -68,7 +68,11 @@ object VolumesModule {
 
   /** @return a validator that checks the validity of a group given the related volume providers */
   def validRootGroup(): Validator[Group] = new Validator[Group] {
-    def apply(grp: Group) =
-      providers.all.map(_.validations.rootGroup).map(validate(grp)(_)).fold(Success)(_ and _)
+    def apply(grp: Group) = {
+      val groupProvider = grp.transitiveApps.flatMap(_.externalVolumes).flatMap {
```
Comment (jdef) [inline src/main/scala/mesosphere/marathon/core/externalvolume/VolumesModule.scala]: does it make sense to `toSet` the providers?

Comment (jdef) [inline src/main/scala/mesosphere/marathon/core/externalvolume/VolumesModule.scala]: nevermind, it's already a set

**Comment by jdef** [comment]
lgtm

File: src/main/scala/mesosphere/marathon/core/externalvolume/VolumesModule.scala (hunk 1 above)
Comment (kolloch) [inline src/main/scala/mesosphere/marathon/core/externalvolume/VolumesModule.scala]: and BTW, that it is a `Set` is a really bad idea ;)

**Comment by kolloch** [comment]
I think that this PR is not necessary...

**Pull request closed without merging**